eva_submission/eload_brokering.py

```diff
@@ -0,0 +1,194 @@
+
+        # Upload XML to ENA
+
+    def upload_vcf_files_to_ena_ftp(self):
+        print('Connect to %s' % cfg.query('ena', 'ftphost'))
+        ftps = HackFTP_TLS()
+        host, port = cfg.query('ena', 'ftphost').split(':')
+        ftps.connect(host, port=int(cfg.query('ena', 'ftpport', ret_default=21)))
+        ftps.login(cfg.query('ena', 'username'), cfg.query('ena', 'password'))
+        ftps.prot_p()
+        if self.eload not in ftps.nlst():
+            self.info('Create %s directory' % self.eload)
+            ftps.mkd(self.eload)
+        ftps.cwd(self.eload)
+        for vcf_file in self.eload_cfg['brokering']['vcf_files']:
+            vcf_file_name = os.path.basename(vcf_file)
+            self.info('Upload %s to FTP' % vcf_file_name)
+            with open(vcf_file, 'rb') as open_file:
+                ftps.storbinary('STOR %s' % vcf_file_name, open_file)
+            vcf_file_index = self.eload_cfg['brokering']['vcf_files'][vcf_file]['index']
+            vcf_file_index_name = os.path.basename(vcf_file_index)
+            self.info('Upload %s to FTP' % vcf_file_index_name)
+            with open(vcf_file_index, 'rb') as open_file:
+                ftps.storbinary('STOR %s' % vcf_file_index_name, open_file)
+
+    def upload_xml_files_to_ena(self, submission_file, project_file, analysis_file):
+        response = requests.post(
+            cfg.query('ena', 'submit_url'),
+            files=dict(SUBMISSION=submission_file, PROJECT=project_file, ANALYSIS=analysis_file)
+        )
+        # TODO: complete the parsing of the response
+
+    def upload_to_bioSamples(self, metadata_spreadsheet):
+        sample_tab_submitter = SampleMetadataSubmitter(metadata_spreadsheet)
+        sample_name_to_accession = sample_tab_submitter.submit_to_bioSamples()
+        self.eload_cfg['brokering']['Biosamples'] = sample_name_to_accession
+
+    def _run_brokering_prep_workflow(self):
+        output_dir = self.create_temp_output_directory()
+        brokering_config = {
+            'vcf_files': self.eload_cfg['validation']['valid']['vcf_files'],
+            'output_dir': output_dir,
+            'executable': cfg['executable']
+        }
+        # run the validation
+        brokering_confg_file = os.path.join(self.eload_dir, 'brokering_confg_file.yaml')
+        with open(brokering_confg_file, 'w') as open_file:
+            yaml.safe_dump(brokering_config, open_file)
+        validation_script = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'nextflow', 'prepare_brokering.nf')
+        try:
+            command_utils.run_command_with_output(
+                'Start Nextflow brokering preparation process',
+                ' '.join((
+                    cfg['executable']['nextflow'], validation_script,
+                    '-params-file', brokering_confg_file,
+                    '-work-dir', output_dir
+                ))
+            )
+        except subprocess.CalledProcessError as e:
```

Comment:
Have you checked if the CalledProcessError gets thrown for errors in Nextflow pipeline executions? Are the exit code semantics similar to a bash command?

Reply:
`CalledProcessError ` get thrown by run_command_with_output when the exit status is non 0. Nextflow definitely exit with non 0 code when it fails for any reason.